File: src/prod/src/Management/RepairManager/NodeImpactLevel.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace Common;

namespace Management
{
    namespace RepairManager
    {
        namespace NodeImpactLevel
        {
            BEGIN_ENUM_TO_TEXT( NodeImpactLevel )
                ADD_ENUM_TO_TEXT_VALUE ( Invalid )
                ADD_ENUM_TO_TEXT_VALUE ( None )
                ADD_ENUM_TO_TEXT_VALUE ( Restart )
                ADD_ENUM_TO_TEXT_VALUE ( RemoveData )
                ADD_ENUM_TO_TEXT_VALUE ( RemoveNode )
            END_ENUM_TO_TEXT( NodeImpactLevel )

            ENUM_STRUCTURED_TRACE( NodeImpactLevel, Invalid, static_cast<Enum>(LAST_ENUM_PLUS_ONE - 1) );

            ErrorCode FromPublicApi(FABRIC_REPAIR_NODE_IMPACT_LEVEL publicValue, Enum & internalValue)
            {
                switch (publicValue)
                {
                case FABRIC_REPAIR_NODE_IMPACT_LEVEL_INVALID:
                    internalValue = Invalid;
                    break;
                case FABRIC_REPAIR_NODE_IMPACT_LEVEL_NONE:
                    internalValue = None;
                    break;
                case FABRIC_REPAIR_NODE_IMPACT_LEVEL_RESTART:
                    internalValue = Restart;
                    break;
                case FABRIC_REPAIR_NODE_IMPACT_LEVEL_REMOVE_DATA:
                    internalValue = RemoveData;
                    break;
                case FABRIC_REPAIR_NODE_IMPACT_LEVEL_REMOVE_NODE:
                    internalValue = RemoveNode;
                    break;
                default:
                    return ErrorCodeValue::InvalidArgument;
                }

                return ErrorCode::Success();
            }

            FABRIC_REPAIR_NODE_IMPACT_LEVEL ToPublicApi(Enum const internalValue)
            {
                switch (internalValue)
                {
                case None:
                    return FABRIC_REPAIR_NODE_IMPACT_LEVEL_NONE;
                case Restart:
                    return FABRIC_REPAIR_NODE_IMPACT_LEVEL_RESTART;
                case RemoveData:
                    return FABRIC_REPAIR_NODE_IMPACT_LEVEL_REMOVE_DATA;
                case RemoveNode:
                    return FABRIC_REPAIR_NODE_IMPACT_LEVEL_REMOVE_NODE;
                default:
                    return FABRIC_REPAIR_NODE_IMPACT_LEVEL_INVALID;
                }
            }
        }
    }
}

File: src/prod/src/managed/Api/src/System/Fabric/Repair/NodeImpactLevel.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Repair
{
    using System.Fabric.Interop;

    /// <summary>
    /// <para>Enumerates the effective impact that a repair is expected to have on a particular node.</para>
    /// <para>This type supports the Service Fabric platform; it is not meant to be used directly from your code.</para>
    /// </summary>
    public enum NodeImpactLevel
    {
        /// <summary>
        /// <para>Indicates that the node impact level is invalid.</para>
        /// </summary>
        Invalid = NativeTypes.FABRIC_REPAIR_NODE_IMPACT_LEVEL.FABRIC_REPAIR_NODE_IMPACT_LEVEL_INVALID,

        /// <summary>
        /// <para>Indicates that no impact is expected.</para>
        /// </summary>
        None = NativeTypes.FABRIC_REPAIR_NODE_IMPACT_LEVEL.FABRIC_REPAIR_NODE_IMPACT_LEVEL_NONE,

        /// <summary>
        /// <para>Indicates that the node is expected to stop and eventually restart.</para>
        /// </summary>
        Restart = NativeTypes.FABRIC_REPAIR_NODE_IMPACT_LEVEL.FABRIC_REPAIR_NODE_IMPACT_LEVEL_RESTART,

        /// <summary>
        /// <para>Indicates that the node is expected to stop, and may lose all of its persisted state prior to restarting.</para>
        /// </summary>
        RemoveData = NativeTypes.FABRIC_REPAIR_NODE_IMPACT_LEVEL.FABRIC_REPAIR_NODE_IMPACT_LEVEL_REMOVE_DATA,

        /// <summary>
        /// <para>Indicates that the node is to be removed from the cluster.</para>
        /// </summary>
        RemoveNode = NativeTypes.FABRIC_REPAIR_NODE_IMPACT_LEVEL.FABRIC_REPAIR_NODE_IMPACT_LEVEL_REMOVE_NODE,
    }
}